task_templates/1_transforms/3_python3_sklearn_transform/custom.py

```diff
@@ -9,56 +9,40 @@
-):
-    """
-    This hook must be implemented with your fitting code, for running drum in the fit mode.
-
-    This hook MUST ALWAYS be implemented for custom tasks. For custom transformers, the
-    transform hook below is also required.
-
-    For inference models, this hook can stick around unimplemented, and won’t be triggered.
-
-    Parameters
-    ----------
-    X: pd.DataFrame - training data to perform fit on
-    y: pd.Series - target data to perform fit on
-    output_dir: the path to write output. This is the path provided in '--output' parameter of the
-        'drum fit' command.
-    kwargs: Added for forwards compatibility
-
-    Returns
-    -------
-    Nothing
-    """
-    transformer = make_pipeline()
-    transformer.fit(X, y)
-
-    # You must serialize out your transformer to the output_dir given, however if you wish to change this
-    # code, you will probably have to add a load_model method to read the serialized model back in
-    # When prediction is done.
-    # Check out this doc for more information on serialization https://github.com/datarobot/custom-\
-    # model-templates/tree/master/custom_model_runner#python
-    # NOTE: We currently set a 10GB limit to the size of the serialized model or transformer
-    with open("{}/artifact.pkl".format(output_dir), "wb") as fp:
-        pickle.dump(transformer, fp)
-
-
-def transform(X, transformer):
-    """
-    Parameters
-    ----------
-    X: pd.DataFrame - training data to perform transform on
-    transformer: object - trained transformer object
-    Returns
-    -------
-    transformed DataFrame resulting from applying transform to incoming data
-    """
-    transformed = transformer.transform(X)
-    if type(transformed) == csr_matrix:
-        return pd.DataFrame.sparse.from_spmatrix(transformed)
-    else:
-        return pd.DataFrame(transformed)
+from datarobot_drum.custom_task_interfaces import TransformerInterface
+
+
+class CustomTask(TransformerInterface):
+    def fit(self, X: pd.DataFrame, y: pd.Series, **kwargs,) -> None:
+        """
+        This hook defines how DataRobot will train this task. Even transform tasks need to be trained to learn/store information from training data
+        DataRobot runs this hook when the task is being trained inside a blueprint.
+        The input parameters are passed by DataRobot based on project and blueprint configuration.
+
+        Parameters
+        ----------
+        X: pd.DataFrame - training data to perform fit on
+        y: pd.Series - target data to perform fit on
+        kwargs: Added for forwards compatibility
+
+        Returns
+        -------
+        None
+        """
+        self.transformer = make_pipeline()
+        self.transformer.fit(X, y)
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Parameters
+        ----------
+        X: pd.DataFrame - training data to perform transform on
+        Returns
```

Comment:
Docstring formatting nit:
```suggestion
        X: pd.DataFrame - training data to perform transform on

        Returns
```